src/CppUTest/TestRegistry.cpp

```diff
@@ -72,6 +72,32 @@ void TestRegistry::runAllTests(TestResult& result)
     currentRepetition_++;
 }
 
+void TestRegistry::listTestGroupNames(TestResult& result)
+{
+    SimpleString lastgroup = "";
+
+    for (UtestShell *test = tests_; test != NULL; test = test->getNext()) {
+        SimpleString gname = test->getGroup();
+        if (gname != lastgroup) {
```

Comment:
This sucks :) @arstrube more prove we need to make TestGroup in a separate thing.